verisignmdns/src/main/java/denominator/verisignmdns/VrsnMdnsContentHandler.java

```diff
@@ -0,0 +1,167 @@
+        private boolean domainElementFound = false; // flag for getting the
+                                                    // value...
+
+        @Override
+        public List<Zone> result() {
+            return zones;
+        }
+
+        @Override
+        public void startElement(String uri, String localName, String qName, Attributes attrs) {
+
+            if (qName != null && qName.endsWith(":domainName")) {
+                domainElementFound = true;
+            }
+        }
+
+        @Override
+        public void characters(char ch[], int start, int length) throws SAXException {
+            if (domainElementFound) {
+                zones.add(Zone.create(new String(ch, start, length)));
+            }
+        }
+
+        @Override
+        public void endElement(String uri, String localName, String qName) throws SAXException {
+            if (qName != null && qName.endsWith(":domainName")) {
+                domainElementFound = false;
+            }
+        }
+    }
+
+    static class RecordListHandler extends DefaultHandler implements ContentHandlerWithResult<List<Record>> {
+        private final List<Record> rrs = new ArrayList<Record>();
+
+        @Inject
+        RecordListHandler() {
+        }
+
+        private Record rr = new Record();
+        private boolean processingRR = false; // flag indicating currently
+                                              // inside resource record
+                                              // element..
+        private String currentElementName = "";
+        private String rDataString = "";
+
+        @Override
+        public List<Record> result() {
+            return rrs;
+        }
+
+        @Override
+        public void startElement(String uri, String localName, String qName, Attributes attrs) {
+
+            if (qName.endsWith(":resourceRecord")) {
+                rr = new Record();
+                processingRR = true;
+                currentElementName = ":resourceRecord";
+            }
+            if (processingRR && qName.endsWith(":resourceRecordId")) {
+                currentElementName = ":resourceRecordId";
+            }
+            if (processingRR && qName.endsWith(":owner")) {
+                currentElementName = ":owner";
+            }
+            if (processingRR && qName.endsWith(":type")) {
+                currentElementName = ":type";
+            }
+            if (processingRR && qName.endsWith(":ttl")) {
+                currentElementName = ":ttl";
+            }
+            if (processingRR && qName.endsWith(":rData")) {
+                currentElementName = ":rData";
+            }
+        }
+
+        @Override
+        public void endElement(String uri, String name, String qName) {
+
+            if (qName.endsWith(":resourceRecord")) {
+                rrs.add(rr);
+                processingRR = false;
+            }
+
+            if (qName.endsWith(":rData")) {
+                String[] tempArray = rDataString.split(",");
```

Comment:
when you find yourself writing code like this, look in Util.

https://github.com/Netflix/denominator/blob/master/model/src/main/java/denominator/common/Util.java#L72